fireapp/lib/pages/settings/setting_view_model.dart

```diff
@@ -2,16 +2,29 @@ import 'dart:async';
 
 import 'package:fireapp/domain/repository/authentication_repository.dart';
 import 'package:fireapp/domain/repository/notification_fcm_token_repository.dart';
-import 'package:firebase_messaging/firebase_messaging.dart';
-import 'package:flutter/cupertino.dart';
+import 'package:fireapp/domain/request_state.dart';
+import 'package:fireapp/pages/settings/setting_navigation.dart';
+import 'package:fireapp/presentation/fireapp_view_model.dart';
 import 'package:injectable/injectable.dart';
+import 'package:rxdart/rxdart.dart';
+import 'package:firebase_messaging/firebase_messaging.dart';
 
+import '../../global/di.dart';
 
 @injectable
-class SettingViewModel {
+class SettingViewModel extends FireAppViewModel
+    implements NavigationViewModel<SettingNavigation> {
   final AuthenticationRepository _authenticationRepository;
   final NotificationFCMTokenRepository _fcmTokenRepository;
 
+  final BehaviorSubject<RequestState<void>> _state =
```

Comment:
I would name this `_signOutState` (and update the name of the other variable too). Theoretically this viewModel would control several independent operations, and it would therefore be better to have a state for each one.